### PR TITLE
Fix a corner case in `repeat`.

### PR DESCRIPTION
**Pull request by midfield**
`jnp.repeat(jnp.array(0), 1, axis=0)` throws an error whereas `np.repeat(np.array(0), 1, axis=0) => np.array([0])`.

### CODE REVIEW

**Comment by midfield** [comment]
I spent some time trying to understand the unit test framework to get the desired test to run, but I couldn't get it to work.  (I tried adding `(1,)` to `non_empty_scalar_shapes`, e.g.

```
nonempty_nonscalar_array_shapes = [(1,), (4,), (3, 4), (3, 1), (1, 4), (2, 1, 4), (2, 3, 4)] 
```

but that didn't seem to work.)

**Comment by jakevdp** [comment]
The relevant test case here is ``shape=()``, which is part of ``nonempty_array_shapes``: https://github.com/google/jax/blob/510af1de64e3a1390f9d6c9d983a62e64045d761/tests/lax_numpy_test.py#L48

The shape is already included in the test suite, but only with ``axis=None``: https://github.com/google/jax/blob/510af1de64e3a1390f9d6c9d983a62e64045d761/tests/lax_numpy_test.py#L1116-L1118

You could change that line to
```python
for axis in [None] + list(range(-len(shape), max(1, len(shape))))))
```
and I believe this case would be covered.


    

```

**Comment by midfield** [comment]
@jakevdp Whoops, that was a goofup, thanks for pointing that out!  I shouldn't do work late Fridays...I added the test but unfortunately with the default settings (`JAX_NUM_GENERATED_CASES=25`) it doesn't seem to trigger.  (Don't know what you folks are using for CI tests.)

BTW out of curiosity why don't you folks use hypothesis?  This issue was found in the TFP testsuite via hypothesis.

**Comment by jekbradbury** [comment]
We currently use 25 for public CI, but significantly more on Google-internal infrastructure. So I think this should be fine; it'll get tested frequently enough. I'm not sure about hypothesis; it does seem pretty nice!
